src/Serializer.php

```diff
@@ -208,4 +220,37 @@ public function getLoader(): NormalizerLoaderInterface
     {
         return $this->loader;
     }
+
+    /**
+     * Create a new denormalization context, overriding the default context
+     *
+     * @param array<string, mixed> $context
+     * @return DenormalizationContext
+     */
+    private function denormalizationContext(array $context): DenormalizationContext
+    {
+        if ($this->defaultDenormalizationOptions !== null) {
+            $context += $this->defaultDenormalizationOptions;
+        }
+
+        return new DenormalizationContext($this, $context);
```

Comment:
Maybe use the DenormalizationContext::duplicate() method as in the normalization context.

Reply:
This is not necessary, because there is no option alias. And the usage of duplicate has performance impacts.

Reply:
I know but this means the serializer knows how the context manage its options.